packages/test/src/test-workflows.ts

```diff
@@ -130,7 +130,24 @@ async function createWorkflow(
 
 async function activate(t: ExecutionContext<Context>, activation: coresdk.workflow_activation.IWorkflowActivation) {
   const { workflow, runId } = t.context;
-  const completion = await workflow.activate(activation);
+
+  // Core guarantees the following jobs ordering: patches -> signals+update -> others
+  // Tests are likely to fail if we artifically pass an activation that does not follow that order
+  const jobs: coresdk.workflow_activation.IWorkflowActivationJob[] = activation.jobs ?? [];
+  function getPriority(job: coresdk.workflow_activation.IWorkflowActivationJob) {
+    if (job.notifyHasPatch) return 0;
+    if (job.signalWorkflow || job.queryWorkflow) return 1;
```

Comment:
It looks like you meant `doUpdate` instead of `queryWorkflow` here?

Reply:
Indeed, thanks